### PR TITLE
GH-22: Add support for coverage build

### PR DESCRIPTION
**Pull request by rfernandes**
Enable coverage builds and report generation.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -156,6 +156,9 @@ endif()
 
 enable_testing()
 
+# Add option to enable coverage builds
+option(TOOLBOX_COVERAGE "Enable profiling and coverage report analysis" OFF)
```
Comment (marayl) [inline CMakeLists.txt]: Should we put this alongside the other options at the top of the file?

Comment (rfernandes) [reply CMakeLists.txt]: Done

File: toolbox/util/Utility.cpp [hunk 2]
```diff
@@ -46,6 +46,25 @@ int hex_digits(int64_t i) noexcept
     return n;
 }
 
+int hex_digits2(int64_t i) noexcept
```
Comment (marayl) [inline toolbox/util/Utility.cpp]: Shall we just replace the original with this version?

**Comment by marayl** [comment]
We should probably agree on a consistent commit message style and update `CONTRIBUTING.md` accordingly. Not one for this PR.

File: toolbox/util/Utility.cpp (hunk 2 above)
Comment (marayl) [reply toolbox/util/Utility.cpp]: Also: have you tried running `make clang-format` on this?

Comment (rfernandes) [reply toolbox/util/Utility.cpp]: Oppps this isn't suppost to be part of this PR. My bad

File: toolbox/CMakeLists.txt [hunk 3]
```diff
@@ -151,6 +151,38 @@ install(FILES contrib/http_parser.h
   COMPONENT header
 )
 
+if(${CMAKE_CXX_COMPILER_ID} MATCHES GNU AND ${TOOLBOX_COVERAGE})
+
+  message(STATUS "Enabling profile build for tests")
+  set_target_properties(${toolbox_LIBRARY}
+    PROPERTIES
+      COMPILE_FLAGS "${CMAKE_CXX_FLAGS} -g -O0 -fprofile-arcs -ftest-coverage --coverage"
```
Comment (marayl) [inline toolbox/CMakeLists.txt]: An alternative to a coverage option would be a new build type. I often use this approach for perf builds and the likes. For example:

```
set(CMAKE_C_FLAGS_PERF   "${CMAKE_C_FLAGS_RELEASE} -fno-omit-frame-pointer")
set(CMAKE_CXX_FLAGS_PERF "${CMAKE_CXX_FLAGS_RELEASE} -fno-omit-frame-pointer")
```

Comment (rfernandes) [reply toolbox/CMakeLists.txt]: Yes it doesn't really make much sense to do coverage on release builds for example, I'll change it to a new build type.

**Pull request merged**